docs/source/conf.py

```diff
@@ -15,24 +15,42 @@
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration
 
 extensions = [
+    'sphinx.ext.intersphinx',
     'sphinx_ansible_theme',
-    'sphinxcontrib.apidoc'
+    'sphinx_immaterial',
```

Comment:
Looks like this extension is auto-injecting their integrated theme which is what's probably causing those them option warnings:
```console
WARNING: unsupported theme option 'font' given
WARNING: unsupported theme option 'plugins' given
WARNING: unsupported theme option 'icon' given
WARNING: unsupported theme option 'repo_url' given
WARNING: unsupported theme option 'edit_uri' given
WARNING: unsupported theme option 'toc_title' given
WARNING: unsupported theme option 'toc_title_is_page_title' given
```
_(copied from the CI output)_

It is probably doing too much and maybe it's possible to only initialize the necessary bits by loading separate sub-extensions, I'm not sure but it's worth a try.

Reply:
https://jbms.github.io/sphinx-immaterial/apidoc/python/apigen.html#usage doesn't say that adding `sphinx_immaterial` is required there. So maybe dropping this would fix some problems:
```suggestion
```

Reply:
Dropping 'spinx_immaterial' leads to errors such as:

```
Sphinx parallel build error:
ValueError: Failed to document entity: 'ansible_sdk.AnsibleJobDef'
```
I'd like to see if there is another way to resolve those unsupported theme option warnings.

Reply:
That error is coming from https://github.com/jbms/sphinx-immaterial/blob/21f2aeeb7fd182955a78893c30aaa6b84d6d5e8e/sphinx_immaterial/apidoc/python/apigen.py#L628.

I'd maybe try having a simple `setup()` function in this `conf.py` with just these: https://github.com/jbms/sphinx-immaterial/blob/a944f27/sphinx_immaterial/__init__.py#L254-L267 to see if it initializes enough internals for this to work.